// react-native-pin-input-component/examples/src/HomeScreen.js
/* eslint-disable react/prefer-stateless-function */

import React from 'react';
import {SafeAreaView, StyleSheet, View, Button} from 'react-native';
import PropTypes from 'prop-types';

import {Colors} from 'react-native/Libraries/NewAppScreen';

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.lighter,
    justifyContent: 'center',
  },
});
class HomeScreen extends React.Component {
  render() {
    const {navigation} = this.props;
    return (
      <SafeAreaView>
        <View style={styles.container}>
          <Button
            title="Default PIN"
            onPress={() => {
              navigation.navigate('default');
            }}
          />
          <Button
            title="Visible Selection"
            onPress={() => {
              navigation.navigate('visibleSelection');
            }}
          />
          <Button
            title="Customizable Style #1"
            onPress={() => {
              navigation.navigate('customizable');
            }}
          />
          <Button
            title="Customizable Style #2"
            onPress={() => {
              navigation.navigate('customizable2');
            }}
          />
          <Button
            title="Customizable View"
            onPress={() => {
              navigation.navigate('customView');
            }}
          />
          <Button
            title="Default Password"
            onPress={() => {
              navigation.navigate('defaultPassword');
            }}
          />
        </View>
      </SafeAreaView>
    );
  }
}

HomeScreen.propTypes = {
  navigation: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default HomeScreen;
