// react-native-pin-input-component/src/Dot.js
import React from 'react';
import { Animated } from 'react-native';
import PropTypes from 'prop-types';

class Dot extends React.Component {
  constructor() {
    super();
    this.state = { scale: new Animated.Value(0) };
  }

  render() {
    const { scale } = this.state;
    const {
      size,
      color,
      onBlurColor,
      onBlur,
      margin,
      animatedOnChangeText,
      resetAnimatied,
    } = this.props;
    const backgroundColor = onBlur ? onBlurColor : color;
    const s = animatedOnChangeText(scale);
    if (!onBlur) resetAnimatied(scale);
    return (
      <Animated.View
        style={{
          width: size,
          height: size,
          backgroundColor,
          margin,
          borderRadius: size / 2,
          transform: [{ scale: s }],
        }}
      />
    );
  }
}

Dot.propTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
  onBlurColor: PropTypes.string,
  onBlur: PropTypes.bool,
  margin: PropTypes.number,
  animatedOnChangeText: PropTypes.func,
  resetAnimatied: PropTypes.func,
};

Dot.defaultProps = {
  size: 12,
  color: '#B2B2B2',
  onBlurColor: 'gray',
  onBlur: false,
  margin: 5,
  animatedOnChangeText: () => {
    return 1;
  },
  resetAnimatied: scale => {
    scale.setValue(0);
  },
};

export default Dot;
